src/client/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs

```diff
@@ -127,6 +127,16 @@ public string Claims
         // TODO: ideally, this can come from the particular request instance and not be in RequestBase since it's not valid for all requests.
         public bool SendX5C { get; set; }
 
+        private string _ccsRoutingHint;
+        /// <summary>
+        /// OID and Tenant ID of the signed-in user for CCS routing.
+        /// </summary>
+        public string CcsRoutingHint 
```

Comment:
This is a bit tricky. I think I understand what you are doing but it's not obvious. `_ccsRoutingHint` is the one from the app developer. `_commonParameters.CcsRoutingHint` is the one that gets calculated from the users right?

I'd maybe rename `_ccsRoutingHint ` with `_userCcsRoutingHint` or smth like that. And not use `commonParameters.CcsRoutingHint` here at all, since it's not technically a request param.